src/cpu/instr.rs

```diff
@@ -1,7 +1,40 @@
 use crate::cpu::cpu::CPU;
+use bitflags::bitflags;
 
 use std::collections::HashMap;
 
+bitflags! {
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    pub struct Flags: u16 {
+        const CLEAR_FLAG = 0;
+        const Z = (1 << 0); //zero
+        const C = (1 << 1); //cary
+        const N = (1 << 2); //negative
+        const O = (1 << 3); //overflow
+        const P = (1 << 4); //parity
+    }
+}
+
+pub fn convert_to_bitflags(mut field: u16) -> Flags {
+    let mut temp_flag: Flags = Flags::CLEAR_FLAG;
+    if field & 0b1 == 1 { temp_flag |= Flags::Z; }
+    field = field >> 1;
+
+    if field & 0b1 == 1 { temp_flag |= Flags::C; }
+    field = field >> 1;
+
+    if field & 0b1 == 1 { temp_flag |= Flags::N; }
+    field = field >> 1;
+
+    if field & 0b1 == 1 { temp_flag |= Flags::O; }
+    field = field >> 1;
+
+    if field & 0b1 == 1 { temp_flag |= Flags::P; }
+    field = field >> 1;
+
+    return temp_flag;
+}
+
```

Comment:
I think that this function is not needed, bitflags provide `fn from_bits(bits: Self::Bits -> Option<Self>`